Batch File/tests/syntax_test_batch_file.bat

```diff
@@ -1408,36 +1434,49 @@ put arg1 arg2
 ::                   ^ keyword.operator.assignment.redirection.dosbatch
 
    ECHO >> NUL
+:: ^^^^^^^^^^^ meta.command.echo.dosbatch
 ::     ^ - meta.redirection
 ::      ^^^^^^ meta.redirection.dosbatch
 ::            ^ - meta.redirection
 ::      ^^ keyword.operator.assignment.redirection.dosbatch
 ::         ^^^ constant.language.null.dosbatch
 
-   dir > f.txt 2>&1
+   ECHO <contet.txt and others
+:: ^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.command.echo.dosbatch
+::      ^^^^^^^^^^^ meta.string.dosbatch meta.redirection.dosbatch - string string
+
+   ECHO "<contet.txt and others"
+:: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.command.echo.dosbatch
+::      ^^^^^^^^^^^^^^^^^^^^^^^^ meta.string.dosbatch string.unquoted.dosbatch - meta.interpolation - meta.redirection
```

Comment:
Not `string.quoted.double`?

Reply:
No, because the output is `"<contet.txt and others"`. Quotes are treated literal in this situation. I know, it's weird. I tried to scope those quotes `punctuation` which are not rendered in output, to make it visible.

The main issue with those old shell languages (also Bash), is that every token / identifier may consist of quoted and unquoted parts, which are chained together. There is basically no such concept of pure quoted or unquoted strings.

Reply:
Understood. _Yuck!_